### PR TITLE
Upgraded dependencies

### PR DESCRIPTION
**Pull request by lil5**
Fix Hjson had an issue with variable types

Upgraded the following dependencies;
cson, fs-extra, hjson, js-yaml, json5, toml-j0.4, tomlify-j0.4

### CODE REVIEW

File: index.js
Comment (MarkTiedemann) [inline index.js]: The comment says `Remove '.' in file extensions`, the name of the function is `validType` -- that's kind of confusing. I'm a big fan of naming functions in such a way that you immediately know what they are doing. So I think something like `removeFileExt` would be a good name.

Also the code can be simplified: 
```js
const removeFileExt = type =>
    type.startsWith('.') ? type.substr(1) : type
```

File: index.js
Comment (MarkTiedemann) [inline index.js]: Instead of
```js
type = validType(type)
switch (type) { ... }
```
do
```js
switch(validType(type)) { ... }
```
please. :)

File: index.js
Comment (MarkTiedemann) [inline index.js:25]: Can you explain why the `string.toString()` is necessary, please? Seems kind of odd to me.

File: index.js
Comment (MarkTiedemann) [inline index.js]: Same as above.

**Review by MarkTiedemann — CHANGES_REQUESTED** [review]

**Comment by MarkTiedemann** [comment]
Thanks for the PR, @lil5!

The CI failure is not related to your code but due to the short timeout of the test command. (I'm gonna fix this as soon as the PR is merged.)

If you want, please also add test cases for the changes that you introduced. For example, to test that the types can have a leading `.` now:

```js
// in test/cli/convert.js line 41

const types = ['json', 'json5', 'hjson', 'cson', 'yaml', 'toml', 'ini']
const extensions = types.concat(types.map(type => '.' + type))
```

Also, it would be great if you could update the API section of the `README.md`. (Otherwise I'm gonna do that later.)

Again, thanks for your PR!

PS: If you want to be added to this project as a maintainer, let me know.

File: index.js
Comment (lil5) [reply index.js]: sure will add that to my next commit

File: index.js
Comment (lil5) [reply index.js]: Were there in the future, to be more code in between, wouldn't it be more readable by stating per line an action instead of having as many actions in one line as possible, minifiers are there for that purpose

File: index.js
Comment (lil5) [reply index.js:25]: Yeah that was a strange one,

using a js file I tested `typeof string` and it returned `Object`, very strange

so I added a simple `.toString`

File: index.js
Comment (lil5) [reply index.js]: 97792e01985efcfc0140ca33f8729abf8bcaf572

File: index.js
Comment (MarkTiedemann) [inline index.js:50]: Please don't export the `removeFileExt` function. That's an implementation detail that should be of no concern for people using this module.

File: package.json
Comment (MarkTiedemann) [inline package.json:11]: Thanks for removing the timeout. :)

File: index.js
Comment (MarkTiedemann) [reply index.js:25]: Is there a way I can reproduce the issue? Otherwise, please remove the line.

If I have time tomorrow, I will investigate a bit more.

File: README.md
Comment (MarkTiedemann) [inline README.md:53]: Please remove the documentation of the `removeFileExt` function, too.

File: index.js
Comment (lil5) [reply index.js:50]: true but how am I going to be able to test it if I can't directly access the function

see: https://github.com/MarkTiedemann/naml/pull/1/commits/38695a165bccd5f4b274eb2e0023acaeba0329e2#diff-f3b199869efde8696563e62cbae0bec6

Comment (MarkTiedemann) [reply index.js:50]: That 1-line function doesn't need to be tested directly I guess. In `test/cli/convert.js` there's a test for all the different types (`json`, `yaml`, etc.), you can just add all the `.`-prefixed types there (`.json`, `.yaml`, etc.). I think that's good enough:

```js
// in test/cli/convert.js line 41

const types = ['json', 'json5', 'hjson', 'cson', 'yaml', 'toml', 'ini']
const extensions = types.concat(types.map(type => '.' + type))
```

Comment (lil5) [reply index.js:50]: Yeah I tried that but it creates files like `input..cson`

Not too hard to change on `ava` but to put effort into removing an optional feature seems counter intuative and it will make the `npm run test` slower as it will need to test on `.cson` and `cson` values

File: index.js
Comment (lil5) [reply index.js:25]: naml/test.js
``` javascript
#!/usr/bin/env node
'use strict'

const HJSON = require('hjson')
const { readFiles } = require('./test/helpers')
const path = require('path')

var file = './test/cli/assets/input.hjson'
console.log('file: '+path.join(process.cwd(), file)+'\n');

readFiles([file])
.then(([string]) => {
  console.log(`\nHJSON.parse(string):`);
  console.log(
    HJSON.parse(string)
  )
})
.catch((err) => {
  console.error(err)
})

readFiles([file])
.then(([string]) => {
  console.log(`\nHJSON.parse(string.toString()):`);
  console.log(
    HJSON.parse(string.toString())
  )
})
.catch((err) => {
  console.error(err)
})


```

./test.js
``` console
file: /media/lil/data/Projects/naml-2/test/cli/assets/input.hjson


HJSON.parse(string):
Error: source is not a string
    at Object.module.exports [as parse] (/media/lil/data/Projects/naml-2/node_modules/hjson/lib/hjson-parse.js:443:39)
    at readFiles.then (/media/lil/data/Projects/naml-2/test.js:15:11)

HJSON.parse(string.toString()):
{ name: 'naml',
  version: '1.0.5',
  description: 'Not Another Markup Language',
  bin: { naml: 'cli.js' },
  main: 'index.js',
  scripts:
   { pretest: 'npm link',
     test: 'nyc ava test/*/*.js --serial --verbose' },
  engines: { node: '>=6.0.0' },
  keywords:
   [ 'naml',
     'json',
     'json5',
     'hjson',
     'cson',
     'yaml',
     'toml',
     'ini',
     'parse',
     'stringify',
     'convert' ],
  author: 'Mark Tiedemann <www.marktiedemann@gmail.com>',
  license: 'WTFPL',
  dependencies:
   { chalk: '^1.1.3',
     cson: '^4.1.0',
     'fs-extra': '^4.0.0',
     hjson: '^3.1.0',
     ini: '^1.3.4',
     'js-yaml': '^3.9.0',
     json5: '^0.5.1',
     'toml-j0.4': '^1.1.0',
     'tomlify-j0.4': '^2.2.0' },
  devDependencies: { ava: '^0.15.2', nyc: '^6.4.4' },
  directories: { test: 'test' },
  repository:
   { type: 'git',
     url: 'git+https://github.com/MarkTiedemann/naml.git' },
  bugs: { url: 'https://github.com/MarkTiedemann/naml/issues' },
  homepage: 'https://github.com/MarkTiedemann/naml' }

```

for more info see:
- `test/helpers.js`
- `test/cli/convert.js`

Comment (MarkTiedemann) [reply index.js:25]: That is super strange, and I can't reproduce it. For me, it logs:

```
λ node test                                                          
file: C:\dev\node\naml\test\cli\assets\input.hjson                   
                                                                     
                                                                     
HJSON.parse(string):                                                 
{ name: 'naml',                                                      
  version: '1.0.3',                                                  
  description: 'Not Another Markup Language',                        
  bin: { naml: 'cli.js' },                                           
  // ...         
                                                                     
HJSON.parse(string.toString()):                                      
{ name: 'naml',                                                      
  version: '1.0.3',                                                  
  description: 'Not Another Markup Language',                        
  bin: { naml: 'cli.js' },                                           
  // ...
```

What's your node version? npm version?

File: index.js
Comment (MarkTiedemann) [reply index.js:50]: But the `input..cson` files will be detected as the type `.cson` instead of just `cson` so it should work.

Test speed is also not really a concern. It's already super slow. :)

In any case, I don't think this really needs to be tested. The function is one line, and super simple at that.

File: index.js
Comment (lil5) [reply index.js:25]: node -v
`v7.9.0`

npm -v
`4.2.0`

Comment (MarkTiedemann) [reply index.js:25]: I figured it out. In the test helper, there's this code to convert a buffer to a string: `new String(buffer)`. `typeof new String(buffer)` is `object`, and not `string`. The correct code should be `buffer.toString()`. :)

I'm gonna merge your PR now, then fix that bug.

Thanks again for the PR -- you have truly improved this module! :)

**Pull request merged**
